clang/lib/CodeGen/CodeGenSYCL.cpp

```diff
@@ -45,12 +45,14 @@ void CodeGenModule::EmitSYCLKernelCaller(const FunctionDecl *KernelEntryPointFn,
   llvm::FunctionType *FnTy = getTypes().GetFunctionType(FnInfo);
 
   // Retrieve the generated name for the SYCL kernel caller function
-  const auto *SKEPAttr = KernelEntryPointFn->getAttr<SYCLKernelEntryPointAttr>();
-  assert(SKEPAttr && "Missing sycl_kernel_entry_point attribute");
-  CanQualType KernelNameType = Ctx.getCanonicalType(SKEPAttr->getKernelName());
-  const SYCLKernelInfo &SKI = Ctx.SYCLKernels.at(KernelNameType);
+  const auto *KernelEntryPointAttr =
+      KernelEntryPointFn->getAttr<SYCLKernelEntryPointAttr>();
```

Comment:
Any reason not to preserve the assert that was there?
```suggestion
  const auto *KernelEntryPointAttr =
      KernelEntryPointFn->getAttr<SYCLKernelEntryPointAttr>();
  assert(KernelEntryPointAttr && "Missing sycl_kernel_entry_point attribute");
```

Reply:
I didn't think it was necessary because we enter this function only if the attribute is present.

Reply:
That is true, but the motivation for use of assert is to ensure that such relationships are sustained during code refactoring. In this context, the assert is used as a function precondition (C++ contracts will provide a better way to declare such requirements in the future; I'm on the hook to deliver a presentation on C++ contracts sometime this year). Granted, in this case, if a caller neglects to ensure that the function actually has the attribute, we'll likely get a SIGSEGV, but an assertion failure generally produces a better understanding of what went wrong.